### ISSUE TITLE
Add option(s) to exclude/stub data during export based on meta-data

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1943]
Not sure if should be done as a part of #1928 and #1942 , probably not.

Main use-case is to not export (by default) any file with (annex) metadata having `distribution-restrictions` set.
One way is to make it accept a dictionary (not sure how in command line) with meta-data fields and desired values regex to be excluded, so by default could then be `exclude_meta={'distribution_restrictions': '.*'}`

Additional option could be `exclude_method` with choices `stub` and `strip` (I like stub better for the default).  In case of `stub` (trickier I guess, but should be doable since we are instructing filename within archive anyways) we would need to replace those files with content `Content of the file was not included into the export. Contact original owner` or something like that.

**Comment by mih** [comment]
If I got it right, this should likely go into `export_archive`. Note that with #2115 metadata, as reported by `metadata` is the state at last aggregation. Hence the export plugin likely wants to query for annex tags directly.

**Comment by yarikoptic** [comment]
Yes, atm for the archive (and thus figshare) experts. Eventually though might need to be checked and supported for annex export

**Comment by yarikoptic** [comment]
so I could sleep well knowing we haven't forgotten about this one, assigned it to the 0.10.1 release ;)
